NAS2D/Renderer/Fade.h

```diff
@@ -28,8 +29,8 @@ namespace NAS2D
 
 		SignalSource<>& fadeComplete();
 
-		void fadeIn(unsigned int durationInMilliseconds);
-		void fadeOut(unsigned int durationInMilliseconds);
+		void fadeIn(const std::chrono::milliseconds& fadeTime);
+		void fadeOut(const std::chrono::milliseconds& fadeTime);
```

Comment:
I believe it would be more consistent with existing code to remove the `const&`. Besides, this is basically a wrapped primitive type, so passing by `const&` would likely be less efficient than passing by value.

Reply:
I'm going to argue against this. You should always use `const` whenever and wherever possible, particularly for non-built-in-types (even if you "know" the underlying type is "just" a primitive).

Reply:
This isn't so much about the `const` and it is about the `&`. Passing by reference adds overhead for small types. If you eliminate the `&`, there's also little point in maintaining the `const`. It's often inappropriate to have `const` on it's own in API signatures when passing or returning types by-value.

The whole point of passing by reference is efficiency. It makes no sense in cases that result in an efficiency loss.

----

You may have noticed in a link you provided just the other day that some compiler checks are disabled for primitives and small types that aren't at least twice the native word size.

https://docs.microsoft.com/en-us/cpp/code-quality/c26820?view=msvc-160

> This warning isn't raised for scalars, smart pointers, or views. It's also not raised for types whose size isn't more than twice the platform-dependent pointer size.

There's good reason why they don't warn in that case.

----

Rules shouldn't be blindly followed without understanding why they're there. Doing so leads to [Cargo cult programming](https://en.wikipedia.org/wiki/Cargo_cult_programming).